### PR TITLE
improve error handling, make extractor message and type separate

### PR DESCRIPTION
**Pull request by max-zilla**
- in status update messages, separate type (PROCESSING, ERROR, etc.) from the contents ("Done processing", "Cannot serialize JSON", etc.) so that Clowder can be updated to use those values and clean up UI
- add better error & retry handling and improve those messages
- add a connector.message_process() method to make it easy for users to send arbitrary messages to Clowder UI, add example of its use to wordcount.py

### CODE REVIEW

**Comment by max-zilla** [comment]
older comments: https://opensource.ncsa.illinois.edu/bitbucket/projects/CATS/repos/pyclowder/pull-requests/103/overview

File: pyclowder/connectors.py [hunk 1]
```diff
@@ -923,31 +927,42 @@ def process_messages(self, channel, rabbitmq_queue):
...
-                                  "status": status_report,
                                   "resource": resource,
-                                  "message": message})
-
-    def message_ok(self, resource):
-        super(RabbitMQHandler, self).message_ok(resource)
+                                  "payload": {
+                                      "file_id":      resource["id"],
+                                      "extractor_id": self.extractor_info['name'],
+                                      "status":       "%s: %s" % (status, message),
```
Comment (robkooper) [inline pyclowder/connectors.py:946]: missing status_report

File: pyclowder/connectors.py [hunk 2]
```diff
@@ -923,31 +927,42 @@ def process_messages(self, channel, rabbitmq_queue):
...
+            # TODO: Remove 'status' from payload later and read from message_type and message in Clowder 2.0
             self.messages.append({"type": "status",
-                                  "status": status_report,
                                   "resource": resource,
-                                  "message": message})
-
-    def message_ok(self, resource):
-        super(RabbitMQHandler, self).message_ok(resource)
+                                  "payload": {
+                                      "file_id":      resource["id"],
```
Comment (robkooper) [inline pyclowder/connectors.py:944]: do we need dataset_id?

Comment (robkooper) [reply pyclowder/connectors.py:944]: this is both dataset_id and file_id, should fix this in the future, both in clowder and pyclowder.

File: pyclowder/connectors.py (hunk 1 above)
Comment (max-zilla) [reply pyclowder/connectors.py:939]: added job_id in the payload so that functionality is retained & removed status_report object (payload replaces it)

**Pull request merged**